worker/host.go

```diff
@@ -47,6 +47,28 @@ type (
 		pts  *prices.PricesCache
 		rhp4 *rhp4.Client
 	}
+
+	hostUploadClient struct {
+		hi api.HostInfo
+		cm api.ContractMetadata
+		rk types.PrivateKey
+
+		acc  *worker.Account
+		csr  ContractSpendingRecorder
+		pts  *prices.PriceTables
+		rhp3 *rhp3.Client
+	}
+
+	hostV2UploadClient struct {
+		hi api.HostInfo
+		cm api.ContractMetadata
```

Comment:
I think all you need here is the `ID` of the contract and not the full metadata?